File: src/main/java/be/kdg/teamh/services/HoofdthemaService.java
package be.kdg.teamh.services;

import be.kdg.teamh.entities.Hoofdthema;
import be.kdg.teamh.exceptions.HoofdthemaNotFoundException;

import java.util.List;

public interface HoofdthemaService
{
    List<Hoofdthema> all();
    void create(Hoofdthema hoofdthema);
    Hoofdthema find(int id) throws HoofdthemaNotFoundException;
    void update(int id, Hoofdthema hoofdthema) throws HoofdthemaNotFoundException;
    void delete(int id) throws HoofdthemaNotFoundException;
}

File: src/main/resources/static/app/scripts/services/HoofdthemaService.js
(function (angular) {

    "use strict";

    function HoofdthemaService($http) {

        var exports = {};

        exports.all = function () {

            return $http.get("/api/hoofdthemas").then(function (response) {
                return response.data;
            });

        };

        exports.find = function (id) {

            return $http.get("/api/hoofdthemas/" + id).then(function (response) {
                return response.data;
            });

        };

        exports.create = function (hoofdthema) {

            return $http.post("/api/hoofdthemas", hoofdthema).then(function (response) {
                return response.data;
            });

        };

        exports.update = function (hoofdthema) {

            return $http.put("/api/hoofdthemas/" + hoofdthema.id, hoofdthema).then(function (response) {
                return response.data;
            });

        };

        exports.delete = function (id) {

            return $http.delete("/api/hoofdthemas/" + id).then(function (response) {
                return response.data;
            });

        };

        return exports;

    }

    angular.module("kandoe").factory("HoofdthemaService", HoofdthemaService);

})(window.angular);
